lib/route-helpers.js

```diff
@@ -0,0 +1,229 @@
+'use strict';
+var _ = require('lodash');
+
+exports.squash = function(type, target, source){
+  return _.reduce(Object.keys(source), function(memo, sourceKey){
+    var sourceValue = source[sourceKey];
+    var targetValue = target[sourceKey];
+    var resultValue;
+    switch(type){
+      case '$pushAll':
+      case '$pullAll':
+        resultValue = targetValue ? targetValue.concat(sourceValue) : sourceValue;
+        break;
+      case '$set':
+      case '$pull':
+      case '$push':
+        resultValue = sourceValue;
+        break;
+      default:
+        throw new Error('Unknown operation ' + type);
+    }
+    memo[sourceKey] = resultValue;
+    return memo;
+  }, target);
+};
+
+exports.needsPositionalUpdate = function(parts, model){
+  var fieldName = parts[0];
+  var fieldSchema = model.schema.tree[ fieldName ];
+  return _.isArray( fieldSchema ) &&
+    _.isObject( fieldSchema[ 0 ] ) &&
+    !_.has( fieldSchema[ 0 ], 'ref' );
+};
+
+exports.processReplaceOp = function(op, model){
+  var field = op.path.split('/').slice(3);
+  var value = op.value;
+  var result = {
+    match: {},
+    separate: false,
+    key: '$set',
+    update: {}
+  };
+  var ret = [result];
+
+  if (field.length > 1 && field[0] !== "links") {
+    var needsPositionalUpdate = exports.needsPositionalUpdate(field, model);
+
+    if (needsPositionalUpdate){
+      //happy to assume it's working one-level only for now
+      var subdocid = field[1];
+      var embeddedPath = field[0];
+      result.match[embeddedPath + '._id'] = subdocid; //new ObjectId(subdocid);
```

Comment:
Should not to be backwards compatible with /resource/0/arraylike/:array-index/path